### PR TITLE
feat: implement handler response on proposal execution

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Implements `handlerReponse` in `executeProposal` function which is passed to `ProposalExecution` event.

## Related Issue Or Context
Closes: #166 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: contracts/handlers/ERC1155Handler.sol [hunk 1]
```diff
@@ -83,6 +83,7 @@ contract ERC1155Handler is IHandler, ERCHandlerHelpers, ERC1155Safe, ERC1155Hold
         } else {
             releaseBatchERC1155(tokenAddress, address(this), address(recipientAddress), tokenIDs, amounts, transferData);
         }
+        return abi.encodePacked(recipientAddress, tokenIDs);
```
Comment (lastperson) [inline contracts/handlers/ERC1155Handler.sol:86]:
```suggestion
        return abi.encode(recipientAddress, tokenIDs, amounts);
```

File: contracts/handlers/ERC20Handler.sol [hunk 2]
```diff
@@ -86,6 +86,7 @@ contract ERC20Handler is IHandler, ERCHandlerHelpers, ERC20Safe {
         } else {
             releaseERC20(tokenAddress, address(recipientAddress), convertToExternalBalance(tokenAddress, amount));
         }
+        return abi.encodePacked(recipientAddress, amount);
```
Comment (lastperson) [inline contracts/handlers/ERC20Handler.sol:89]:
```suggestion
        return abi.encode(tokenAddress, recipientAddress, amount);
```

File: contracts/handlers/ERC721Handler.sol [hunk 3]
```diff
@@ -110,6 +110,7 @@ contract ERC721Handler is IHandler, ERCHandlerHelpers, ERC721Safe {
         } else {
             releaseERC721(tokenAddress, address(this), address(recipientAddress), tokenID);
         }
+        return abi.encodePacked(recipientAddress, tokenID);
```
Comment (lastperson) [inline contracts/handlers/ERC721Handler.sol:113]:
```suggestion
        return abi.encode(tokenAddress, recipientAddress, tokenID);
```

File: contracts/handlers/PermissionedGenericHandler.sol [hunk 4]
```diff
@@ -128,7 +128,7 @@ contract PermissionedGenericHandler is IHandler {
         @notice If {_tokenContractAddressToTokenProperties[contractAddress].executeFunctionSignature} is set,
         {metaData} is expected to consist of needed function arguments.
      */
-    function executeProposal(bytes32 resourceID, bytes calldata data) external onlyBridge {
+    function executeProposal(bytes32 resourceID, bytes calldata data) external onlyBridge returns (bytes memory) {
```
Comment (lastperson) [inline contracts/handlers/PermissionedGenericHandler.sol:131]: Perhaps we should return the execution contract response?

File: contracts/handlers/PermissionlessGenericHandler.sol [hunk 5]
```diff
@@ -142,7 +142,7 @@ contract PermissionlessGenericHandler is IHandler {
           After this, the target contract will get the following:
           executeFuncSignature(address executionDataDepositor, uint[] uintArray, address addr)
      */
-    function executeProposal(bytes32 resourceID, bytes calldata data) external onlyBridge {
+    function executeProposal(bytes32 resourceID, bytes calldata data) external onlyBridge returns (bytes memory) {
```
Comment (lastperson) [inline contracts/handlers/PermissionlessGenericHandler.sol:145]: Perhaps we should return the execution contract response?

File: contracts/handlers/XC20Handler.sol [hunk 6]
```diff
@@ -86,6 +86,7 @@ contract XC20Handler is IHandler, ERCHandlerHelpers, XC20Safe {
         } else {
             releaseERC20(tokenAddress, address(recipientAddress), convertToExternalBalance(tokenAddress, amount));
         }
+        return abi.encodePacked(recipientAddress, amount);
```
Comment (lastperson) [inline contracts/handlers/XC20Handler.sol:89]:
```suggestion
        return abi.encode(tokenAddress, recipientAddress, amount);
```

**Review by lastperson — CHANGES_REQUESTED** [review]
1. I think we need to use abi.encode to make the data easily decodeable.
2. Wouldn't it make sense to return all the useful information from the handler if we are at it already?

**Review by mpetrun5 — APPROVED** [review]

**Review by P1sar — APPROVED** [review]

**Review by lastperson — APPROVED** [review]

**Pull request merged**